case-lib/lib.sh

```diff
@@ -343,3 +343,26 @@ disable_kernel_check_point()
 {
     KERNEL_CHECKPOINT="disabled"
 }
+
+is_zephyr()
+{
+    # check if jq is installed, will remove this part
+    # after all DUTs have jq installed.
+    hash jq 2>/dev/null || sudo apt install jq -y &>/dev/null
+
+    manifest=/etc/sof/manifest.txt
+    jq '.version.firmwareType' "$manifest" | grep "zephyr"
```

Comment:
how about adding the OS (XTOS / Zephyr) information to firmware extended manifest?

Reply:
`grep -q`.

I think there's also a `test -e "$manifest" || return 1` missing before the `jq`.

Reply:
> how about adding the OS (XTOS / Zephyr) information to firmware extended manifest?

I'm OK with that, if we all agree, I can add it in deploy.sh.